R/int.R

```diff
@@ -0,0 +1,72 @@
+# Reference: https://github.com/gzuidhof/zarr.js/blob/292804/src/core/slice.ts#L78
+# TODO: for now these are defined but not used, should be used in the future ? 
+
+#' Abstract Int object
+#' @title Int Class
+#' @docType class
+#' @description
+#' Class representing an indexing of a ZARR store
+#' @noRd
+#' @keywords internal
+Int <- R6::R6Class("Int",
+                     public = list(
+                       #' @field start start index
+                       index = NULL,
+                       #' @description Create a new `Int` object
+                       #' @param index integer index
+                       initialize = function(index) {
+                         self$index <- index
+                       },
+                       #' @description
+                       #' This method takes a single integer argument `length_param` and computes information about the
+                       #' slice that the slice object would describe if applied to a sequence of `length_param` items.
+                       #' It returns a tuple of three integers; respectively these are the start and stop indices
+                       # and the step or stride length of the slice. Missing or out-of-bounds indices are handled
+                       # in a manner consistent with regular slices.
+                       #' @param length_param integer length parameter for calculation of integer indices
+                       indices = function(length_param) {
+                         
+                         # check length_param
+                         index <- self$index[self$index <= length_param]
+                         
+                         # return
+                         return(c(index, length_param))
+                       }
+                     )
+)
+
+#' #' Convenience function for the internal Int class constructor.
+#' #' @param index The integer index.
+#' #' @param zero_based The index of the dimension. By default, FALSE for R-like behavior.
+#' #' @return A Int instance with the specified parameters.
+#' #' @export
+#' int <- function(index, zero_based = FALSE) {
+#'   index_offset <- ifelse(zero_based, 0, -1)
+#'   if(!is_na(index) && is.numeric(index)) {
+#'     index <- index + index_offset
+#'   }
+#'   # Assumed to be zero-based
+#'   # and stop-inclusive
+#'   return(Int$new(
+#'     index = index
+#'   ))
+#' }
+#' 
+#' #' Convenience function for the internal Int class constructor 
+#' #' with zero-based indexing 
+#' #' @param index integer index
+#' #' @export
+#' zb_int <- function(index) {
```

Comment:
I think it could make sense to have a `zb_int` function so that users who want to use zero-based / python-style indexing can do so without much thinking. It could just do `index + 1` i think

Reply:
`zb_int` and `int` should be correctly implemented now, added tests for this as well but please let me know if I have missed anything. 

Now, `IntArrayDimIndexer` accepts Python based indexing which is then transformed to `zero-based-to-one-based` later. 

`get_orthogonal_selection` and `manage_filters` now passes random indices as `int(...)` to allow users to use `[` method with R based indexing, but you should use Python-based indexing in `get_item`. See the commits for more information.